### PR TITLE
Lagrede filter fargekategori

### PR DESCRIPTION
**Pull request by slovrid**

### CODE REVIEW

File: src/filtrering/filtrering-status/fargekategori.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    const dispatch = useDispatch();
+    const statusTall = useStatustallVeilederSelector();
+    const filtervalg = usePortefoljeSelector(OversiktType.minOversikt).filtervalg;
+    const ferdigfilter = filtervalg.ferdigfilterListe;
+    const fargekategoriFilter = filtervalg.fargekategorier;
+
+    const hovedfilterChecked =
+        ferdigfilter.includes(MINE_FARGEKATEGORIER) &&
+        alleFargekategoriFilterAlternativer.every(f => fargekategoriFilter.includes(f));
+    const hovedfilterIndeterminate =
```
Comment (dragutin-nav) [inline src/filtrering/filtrering-status/fargekategori.tsx:38]: Need bit more explanation on this line

File: src/ducks/filtrering.ts [hunk 2]
```diff
@@ -124,6 +128,30 @@ export default function filtreringReducer(state: FiltervalgModell = initialState
...
+            const hovedfilterAlleredeValgt = state.ferdigfilterListe.includes(MINE_FARGEKATEGORIER);
+
+            const nyFerdigfilterListe = hovedfilterAlleredeValgt
+                ? state.ferdigfilterListe.filter(f => f !== MINE_FARGEKATEGORIER)
+                : [...state.ferdigfilterListe, MINE_FARGEKATEGORIER];
+            const nyFargekategorier = hovedfilterAlleredeValgt ? [] : [...alleFargekategoriFilterAlternativer];
+
+            return {...state, ferdigfilterListe: nyFerdigfilterListe, fargekategorier: nyFargekategorier};
+        }
+        case FARGEKATEGORIER_UNDERFILTER_KLIKK: {
```
Comment (dragutin-nav) [inline src/ducks/filtrering.ts:141]: what is underfilter?

Comment (slovrid) [reply src/ducks/filtrering.ts:141]: Red = "Hovedfilter"
Green = "Underfilter"

I am open for suggestions for a better naming convention 😄 

Screenshot 2024-03-18 at 10 53 07

Comment (slovrid) [reply src/ducks/filtrering.ts:141]: Screenshot 2024-03-18 at 10 53 07

File: src/filtrering/filtrering-status/fargekategori.tsx (hunk 1 above)
Comment (slovrid) [reply src/filtrering/filtrering-status/fargekategori.tsx:38]: I you look at the image i pasted in the other comment, the "Hovedfilter" can have multiple states; `unchecked`, `checked`, `indeterminate`. The "Underfilter" affects which state the "Hovedfilter" is in.

**Review by dragutin-nav — APPROVED** [review]

**Review by Mathiamu — APPROVED** [review]

**Pull request merged**